src/main/java/org/apereo/portlet/soffit/model/v1_0/Request.java

```diff
@@ -19,60 +19,50 @@
 
 package org.apereo.portlet.soffit.model.v1_0;
 
-import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
+/**
+ * Provides some information about the user's request to which the portal will
+ * respond using output from the soffit.
+ *
+ * @author drewwills
+ */
 public class Request {
 
-    private String Etag;
-    private String windowId;
-    private String namespace;
-    private String authType;
-    private String mode;
-    private String windowState;
-    private String scheme;
-    private String serverName;
-    private int serverPort;
-    private boolean secure;
+    /**
+     * @since 5.0
+     */
+    public static final String MODE = "mode";
```

Comment:
This one is hard.  I dropped the member variable, but kept (downgraded) the concept of Mode as request attribute.  The view selection logic has support for modes, if you wan to use them.  But this logic will happily let you ignore modes altogether... and that's what I expect the majority to do.